tools/c7n_gcp/c7n_gcp/resources/gke.py

```diff
@@ -79,6 +79,71 @@ def augment(self, resources):
                 r['labels'] = r['resourceLabels']
         return resources
 
+@KubernetesCluster.filter_registry.register('effective-firewall')
+class EffectiveFirewall(ValueFilter):
+    """Filters gke clusters  by their effective firewall rules.
+    See `getEffectiveFirewalls
+    https://cloud.google.com/workflows/docs/reference/googleapis/compute/v1/networks/getEffectiveFirewalls`_
+    for valid fields.
+
+    :example:
+
+    Filter all gke clusters that have a firewall rule that allows public
+    access
+
+    .. code-block:: yaml
+
+        policies:
+           - name: find-publicly-accessable-clusters
+             resource: gcp.gke-cluster
+             filters:
+             - type: effective-firewall
+               key: sourceRanges[]
+               op: contains
+               value: "0.0.0.0/0"
+    """
+
+    schema = type_schema('effective-firewall', rinherit=ValueFilter.schema)
+    permissions = ('compute.instances.getEffectiveFirewalls',)
+    annotation_key = "c7n:firewall"
+
+    def get_firewalls(self, client, p, r):
+        def get_port_ranges(ports):
```

Comment:
this looks like the same static function used in #9046, please extract that to a c7n_gcp/utils.py module, neither needs to be defined as a closure.